### PR TITLE
#804: Disable MLS icon when no geometry

### PR DESCRIPTION
**Pull request by dsuren1**
Disable MLS icon when no geometry

### CODE REVIEW

File: frontend/js/components/template/Card.jsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ class Card extends React.Component {
         if (this.props.card.loadingCardTemplateError) {
             return this.renderLoadTemplateException();
         }
-        const mlsButtonDisable = (!includes(window.location.hash, 'map') && isEmpty(this.props?.rowData?.geometry?.coordinates)) || isEmpty(this.props.geometryType);
+        const mlsButtonDisable = isEmpty(this.props?.rowData?.geometry?.coordinates);
```
Comment (mbarto) [inline frontend/js/components/template/Card.jsx:131]: Instead of disabling the button, maybe it is button to now show it at all, what do you think?

**Review by mbarto — CHANGES_REQUESTED** [review]

File: frontend/js/components/template/Card.jsx (hunk 1 above)
Comment (dsuren1) [reply frontend/js/components/template/Card.jsx:131]: Yes, makes sense. I will modify it accordingly

File: frontend/js/components/template/Card.jsx [hunk 2]
```diff
@@ -128,15 +128,16 @@ class Card extends React.Component {
         if (this.props.card.loadingCardTemplateError) {
             return this.renderLoadTemplateException();
         }
-        const mlsButtonDisable = (!includes(window.location.hash, 'map') && isEmpty(this.props?.rowData?.geometry?.coordinates)) || isEmpty(this.props.geometryType);
+        const hideMLSButton = isEmpty(this.props?.rowData?.geometry?.coordinates);
```
Comment (mbarto) [inline frontend/js/components/template/Card.jsx:131]:
```suggestion
        const showMLSButton = this.props.mlsShow && !isEmpty(this.props?.rowData?.geometry?.coordinates);
```

File: frontend/js/components/template/Card.jsx [hunk 3]
```diff
@@ -128,15 +128,16 @@ class Card extends React.Component {
...
 
         const Template = (
             <div className="scheda-sira">
                 <TemplateSira template={this.props.card.template} model={model}/>
                 <div id="card-btn-group">
-                    <Button id="multiLayerSelect" disabled={mlsButtonDisable} style={{display: this.props.mlsShow ? 'inline-block' : 'none'}} onClick={this.onClickMLS}>
+                    {!hideMLSButton && <Button id="multiLayerSelect" style={{display: this.props.mlsShow ? 'inline-block' : 'none'}} onClick={this.onClickMLS}>
```
Comment (mbarto) [inline frontend/js/components/template/Card.jsx:137]:
```suggestion
                    <Button id="multiLayerSelect" style={{display: showMLSButton ? 'inline-block' : 'none'}} onClick={this.onClickMLS}>
```

File: frontend/js/components/template/Card.jsx [hunk 4]
```diff
@@ -128,15 +128,16 @@ class Card extends React.Component {
...
                         <img src={img} width={16} alt=""/>
                     </Button>
+                    }
```
Comment (mbarto) [inline frontend/js/components/template/Card.jsx:140]:
```suggestion
                    
```

**Review by mbarto — CHANGES_REQUESTED** [review]

**Review by mbarto — APPROVED** [review]

**Pull request merged**